chainer/functions/split_axis.py

```diff
@@ -22,10 +21,23 @@ class SplitAxis(function.Function):
     """Function that splits multiple arrays towards the specified axis."""
 
     def __init__(self, indices_or_sections, axis):
+        if not isinstance(indices_or_sections, (int, collections.Iterable)):
+            raise TypeError('indices_or_sections must be integer or 1-D array')
         self.indices_or_sections = indices_or_sections
         self.axis = axis
 
     def forward_cpu(self, x):
+        if isinstance(self.indices_or_sections, collections.Iterable):
+            cdimx = x[0].shape[self.axis]
+            ind = list(self.indices_or_sections)
+            ind.append(cdimx)
+            prev_i = 0
+            for i in ind:
+                cdimy = max(0, min(i, cdimx) - prev_i)
+                if cdimy == 0:
+                    raise ValueError('Not support if shape contains 0')
```

Comment:
More explanatory error is appropriate.
Output tensors whose shapes contain 0 are not supported.